web/src/buddy_list.ts

```diff
@@ -313,6 +313,14 @@ export class BuddyList extends BuddyListConf {
 
         this.update_empty_list_placeholders();
         this.fill_screen_with_content();
+
+        // `populate` always rerenders all user rows, so we add new event
+        // handlers to all of them.
+        $("#user-list .user-profile-picture img")
+            .off("load")
+            .on("load", function (this: HTMLElement) {
+                $(this).closest(".user-profile-picture").toggleClass("avatar_placeholder", false);
+            });
```

Comment:
Does adding an on-load handler do something if the elements are already loaded?

Generally I'm worried this might be great when profiled, so probably we should test that.

Reply:
Either we should check user settings and not call this block if the user list style will never show avatars -- or at least adjust the comment to clarify that this logic only does anything if they have that setting enabled, and otherwise the jQuery selector will always be the empty set.

Reply:
Yeah it seems like a comment explaining that the jQuery selector returns the empty set for cases with no avatars is simplest.

Reply:
Also, for future reference, the profiling results are here: https://chat.zulip.org/#narrow/channel/101-design/topic/userlist.20avatar.20slow.20load.20.2332264/near/1978113